mailbox/backup/src/test/java/org/apache/james/mailbox/backup/task/ExportServiceTest.java

```diff
@@ -0,0 +1,198 @@
+    private static final String PASSWORD = "password";
+    private static final String CORRESPONDING_FILE_HEADER = "corresponding-file";
+    public static final String MESSAGE_CONTENT = "MIME-Version: 1.0\r\n" +
+        "Subject: test\r\n" +
+        "Content-Type: text/plain; charset=UTF-8\r\n" +
+        "\r\n" +
+        "testmail";
+
+    private ExportService testee;
+    private InMemoryMailboxManager mailboxManager;
+    private MailboxSession bobSession;
+    private FakeMailContext mailetContext;
+
+    @BeforeEach
+    void setUp(FileSystem fileSystem) throws Exception {
+        mailboxManager = InMemoryIntegrationResources.defaultResources().getMailboxManager();
+        bobSession = mailboxManager.createSystemSession(BOB);
+
+        DefaultMailboxBackup backup = createMailboxBackup();
+        DNSService dnsService = createDnsService();
+        BlobStore blobStore = new MemoryBlobStore(new HashBlobId.Factory(), new MemoryDumbBlobStore());
+        mailetContext = FakeMailContext.builder().postmaster(MailAddressFixture.POSTMASTER_AT_JAMES).build();
+        BlobExportMechanism blobExport = createBlobExportMechanism(fileSystem, dnsService, blobStore);
+        MemoryUsersRepository usersRepository = createUsersRepository(dnsService);
+
+        testee = new ExportService(backup, blobStore, blobExport, usersRepository);
+
+    }
+
+    private MemoryUsersRepository createUsersRepository(DNSService dnsService) throws Exception {
+        MemoryDomainList domainList = new MemoryDomainList(dnsService);
+        MemoryUsersRepository usersRepository = MemoryUsersRepository.withVirtualHosting(domainList);
+
+
+        domainList.addDomain(DOMAIN);
+        usersRepository.addUser(BOB, PASSWORD);
+        return usersRepository;
+    }
+
+    private BlobExportMechanism createBlobExportMechanism(FileSystem fileSystem, DNSService dnsService, BlobStore blobStore) {
+        return new LocalFileBlobExportMechanism(mailetContext, blobStore, fileSystem, dnsService,
+            LocalFileBlobExportMechanism.Configuration.DEFAULT_CONFIGURATION);
+    }
+
+    private DNSService createDnsService() throws UnknownHostException {
+        InetAddress localHost = mock(InetAddress.class);
+        when(localHost.getHostName()).thenReturn(JAMES_HOST);
+        DNSService dnsService = mock(DNSService.class);
+        when(dnsService.getLocalHost()).thenReturn(localHost);
+        return dnsService;
+    }
+
+    private DefaultMailboxBackup createMailboxBackup() {
+        ArchiveService archiveService = new Zipper();
+        MailArchivesLoader archiveLoader = new ZipArchivesLoader();
+        ZipMailArchiveRestorer archiveRestorer = new ZipMailArchiveRestorer(mailboxManager, archiveLoader);
+        return new DefaultMailboxBackup(mailboxManager, archiveService, archiveRestorer);
+    }
+
+    @Test
+    void exportUserMailboxesDataShouldReturnCompletedWhenUserDoesNotExist() {
```

Comment:
does the caller of the API has any kind of feedback to know something went wrong?